projects/cdk/utils/math/round.ts

```diff
@@ -23,9 +23,19 @@ function calculate(
 
     precision = Math.min(precision, MAX_PRECISION);
 
-    const pair = `${value}e`.split('e');
-    const tempValue = func(Number(`${pair[0]}e${Number(pair[1]) + precision}`));
-    const processedPair = `${tempValue}e`.split('e');
+    const [significand, exponent = ''] = `${value}`.split('e');
+    const roundedInt = func(Number(`${significand}e${Number(exponent) + precision}`));
+
+    /**
+     * TODO: use BigInt after bumping Safari to 14+
+     */
+    ngDevMode &&
+        console.assert(
+            Number.isSafeInteger(roundedInt),
+            'Impossible to correctly round the such large number',
```

Comment:
«the» is not needed, such A large number

Reply:
```suggestion
            'Impossible to correctly round a such large number',
```